image/image.go

```diff
@@ -87,6 +87,16 @@ func validate(w walker, refs []string, out *log.Logger) error {
 		if err := m.validate(w); err != nil {
 			return err
 		}
+
+		ml, err := findManifestList(w, d)
```

Comment:
This seems not correct, `findManifestList(w, d)` here just assume the `d` is a `manifest list`, but actually, `d` could be manifest or manifest list.   This is why the test failed.
````
--- FAIL: TestValidateLayout (0.03s)
	image_test.go:175: blobs/sha256/3b6be0edf862ac9e6e790fb010cf26770d9ab266f3ff81b5b82f0cf8d6f7d1a6: manifestlist validation failed: [manifests: manifests is required]
````
I think we should check the `d` after [`d.validate(w, validRefMediaTypes)`] (https://github.com/opencontainers/image-tools/pull/51/files#diff-da07f373046a0a24e67b18699dc15a70L78), if it is a manifest, then go to `findManifest`, if it is a manifest list, then go to `findManifestList`